### PR TITLE
add the cln-grpc plugin

### PR DESCRIPTION
**Pull request by openoms**
releated #3054

### CODE REVIEW

File: home.admin/config.scripts/cl-plugin.cln-grpc.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  sudo ln -s /home/bitcoin/cl-plugins-available/cln-grpc/debug/cln-grpc \
+   /home/bitcoin/${netprefix}cl-plugins-enabled/
+  #blitz.conf.sh set [key] [value] [?conffile] <noquotes>
+  /home/admin/config.scripts/blitz.conf.sh set grpc-port "7777" ${CLCONF} noquotes
+  /home/admin/config.scripts/blitz.conf.sh set ${netprefix}cln-grpc-port "7777"
+
+elif [ "$1" = off ]; then
+    sed -i "/^grpc-port/d" ${CLCONF}
+    rm -rf /home/bitcoin/${netprefix}cl-plugins-enabled/cln-grpc
+    /home/admin/config.scripts/blitz.conf.sh set ${netprefix}cln-grpc-port "off"
```
Comment (openoms) [inline home.admin/config.scripts/cl-plugin.cln-grpc.sh:38]: @rootzoll question: is there a benefit to use the port number here or should it be just 
`${netprefix}cln-grpc=on|off` in the `raspiblitz.conf`?

Also the port number is completely arbitrary, suggestions are welcome.
For example on the 4*3 phone keyboard GRPC is 4772.

File: home.admin/config.scripts/cl-plugin.cln-grpc.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+if [ "$1" = install ]; then
+  buildGRPCplugin
+
+elif [ "$1" = on ]; then
+  buildGRPCplugin
+  sudo ln -s /home/bitcoin/cl-plugins-available/cln-grpc/debug/cln-grpc \
+   /home/bitcoin/${netprefix}cl-plugins-enabled/
+  #blitz.conf.sh set [key] [value] [?conffile] <noquotes>
+  /home/admin/config.scripts/blitz.conf.sh set grpc-port "7777" ${CLCONF} noquotes
+  /home/admin/config.scripts/blitz.conf.sh set ${netprefix}cln-grpc-port "7777"
```
Comment (openoms) [inline home.admin/config.scripts/cl-plugin.cln-grpc.sh]: @rootzoll question: is there a benefit to use the port number here or should it be just 
`${netprefix}cln-grpc=on|off` in the `raspiblitz.conf` and in redis?

Also the port number is completely arbitrary, suggestions are welcome.
For example on the 4*3 phone keyboard GRPC is 4772.

**Comment by openoms** [comment]
building the `grpc-plugin` takes a good few minutes. I suggest it should be part of the "fatpack" SDcard image and can be on by default just like the experimental-features of CLN.

File: home.admin/config.scripts/cl-plugin.cln-grpc.sh (hunk 2 above)
Comment (rootzoll) [reply home.admin/config.scripts/cl-plugin.cln-grpc.sh]: I mean its nice to have the port numer as an option - the prestart script should take care of always setting it number. Only point is that its a bit more fuzzy as testing against "on" to check if the service shoudl be active. So now we would test if NOT "" - but that should be basically fine.

Comment (rootzoll) [reply home.admin/config.scripts/cl-plugin.cln-grpc.sh]: port numbner 4772 sounds good with me - nice idea with the GRPC phonekeyboard coding :)

**Comment by rootzoll** [comment]
Please merge if you see this PR as done. Intense testing will be done as part of RC1

**Comment by openoms** [comment]
The cln-grpc plugin will be the part of the image release as it will be built after CLN in the build_sdcard.sh. Meging to `dev` for testing.

**Pull request merged**